### ISSUE TITLE
Server API endpoint not called if the URL is missing "randomfilename.png" 

### ISSUE THREAD

**Issue by rushilMethsof** [rs/SDWebImage#2420]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 11.0 / 10.13.0 / 11.0 / 4.0
 SDWebImage Version      | e.g. 4.2.0 / 4.1.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 9 / Xcode 8
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

I can confirm that `sd_setImage` does not hit the server if the url is missing some string that ends with imageformat extension, for example `.png`, `.jpg`

Initially the URL was something like `/path/to/image/{id}`, the id was enough to find and return image. In that case, the library would not hit the server.

However , after modifying the route to accept a fake filename at the end it started working
`/path/to/image/{id}?name=image.jpg`

**Comment by dreampiggy** [comment]
By default, SDWebImage's HTTP request, will use a `NSURLSessionDataTask` to download anything from the URL. It supports any HTTP URL like `https://foo/bar/1`, if the HTTP response is correct and return a valid HTTP content-type. (HTTP response which does not contains `Content-Length` is adoptable, but this will disable the progressive download feature, so it's better to provide this header in HTTP response)

I think, you should better try to use a HTTP debugger tool, like [Charles](https://www.charlesproxy.com/) to identify the real HTTP requset & response. And if you can provide a more detailed information, like using SDWebImage version, iOS firmware version, demo, is better for us to idenfity and provide the suggestion.

**Comment by rushilMethsof** [comment]
you can close this issue if you like, just wanted to help if someone else is facing a similar problem

I used postman to initially get image using the URL "/path/to/image/{id}", it returns the image as expected.The http response and content-type were correct. however the same did not work with SDWebImage
After appending a filename and modifying the route , it worked on SDWebImage.
I am using the latest version of SDWebImage, IOS 11.3 

This is great library, thanks for your continuous effort improving it.

**Comment by zhongwuzw** [comment]
@rushilMethsof Feel free to reopen, we all can help you, would you please give a `url` of image that can reproduce? a little demo would be the best.

**Comment by bpoplauschi** [comment]
Thanks you guys for following up on this, indeed I would like to see a `url` that reproduces this issue, since it is something we should fix (if reproducible).

**Comment by bpoplauschi** [comment]
@rushilMethsof I have re-added the issue template info (which is very important) to the issue description. Please fill in the info there.

**Comment by rushilMethsof** [comment]
because of the nature of the project, I will not be able to post the actual URL here.
I may have to create a dummy local server and allow access to it from my home laptop.
that might take some time

**Comment by bpoplauschi** [comment]
Then what can we do next? @rushilMethsof can you debug and see what is happening? If the request goes through? If we get a response?

**Comment by bpoplauschi** [comment]
Closing due to lack of response. @rushilMethsof feel free to reopen if you are willing to help us reproduce and fix the issue.
